lib/internal/url.js

```diff
@@ -1400,6 +1400,33 @@ function getPathFromURL(path) {
   return isWindows ? getPathFromURLWin32(path) : getPathFromURLPosix(path);
 }
 
+function NativeURL(ctx) {
+  this[context] = ctx;
+}
+util.inherits(NativeURL, URL);
+
+function constructUrl(flags, protocol, username, password,
+                      host, port, path, query, fragment) {
+  var ctx = new URLContext();
+  ctx.flags = flags;
+  ctx.scheme = protocol;
+  ctx.username = username;
+  ctx.password = password;
+  ctx.port = port;
+  ctx.path = path;
+  ctx.query = query;
+  ctx.fragment = fragment;
+  ctx.host = host;
+  const url = new NativeURL(ctx);
+  if (!url[searchParams]) { // invoked from URL constructor
+    url[searchParams] = new URLSearchParams();
+    url[searchParams][context] = this;
```

Comment:
Additionally, since this function is never "invoked from URL constructor" but `!url[searchParams]` is always true, you might want to remove the conditional part.